core/src/main/java/com/linecorp/armeria/common/SuccessFunction.java

```diff
@@ -0,0 +1,84 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import com.linecorp.armeria.client.WebClientBuilder;
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.logging.LoggingService;
+import com.linecorp.armeria.server.metric.MetricCollectingService;
+
+/**
+ * A function that accepts {@link RequestContext} and {@link RequestLog} for checking if the response is
+ * success. Like {@link ServerBuilder} and {@link WebClientBuilder} are using this interface to decide
+ * the response is success or not for Metric and Logging decorator.
+ *
+ * <p>Example:
+ * <pre>{@code
+ * ServerBuilder sb = ServerBuilder
+ *         .successFunction((ctx, req) -> req.responseHeaders().status().code() == 200
+ *                                        || req.responseHeaders().status().code() == 404)
+ *         .decorator(MetricCollectingService.newDecorator(MeterIdPrefixFunction.ofDefault("myServer")))
+ *         .decorator(LoggingService.newDecorator()));
+ *
+ * WebClient client = WebClient
+ *         .builder(uri)
+ *         .successFunction((ctx, req) -> req.responseHeaders().status().code() == 200
+ *                                        || req.responseHeaders().status().code() == 404)
+ *         .decorator(MetricCollectingClient.newDecorator(MeterIdPrefixFunction.ofDefault("myClient")))
+ *         .decorator(LoggingClient.newDecorator()))
+ *         .build();
+ * }
+ * </pre>
+ *
+ */
+@FunctionalInterface
+public interface SuccessFunction {
+    /**
+     * Default success response classification function which checks
+     * {@link RequestLog#responseCause()} is null, 100 &lt;= {@link HttpStatus} &lt; 400
+     * and {@link RpcResponse#isCompletedExceptionally()} == {@code false}.
+     */
+    static SuccessFunction ofDefault() {
+        return (ctx, log) -> {
+            if (log.responseCause() != null) {
+                return false;
+            }
+
+            final int statusCode = log.responseHeaders().status().code();
+            if (statusCode < 100 || statusCode >= 400) {
+                return false;
+            }
+
+            final Object responseContent = log.responseContent();
+            if (responseContent instanceof RpcResponse) {
+                return !((RpcResponse) responseContent).isCompletedExceptionally();
+            }
+
+            return true;
+        };
+    }
+
+    /**
+     * Return true if the response is success.
+     *
+     * @see LoggingService#serve(ServiceRequestContext, HttpRequest)
+     * @see MetricCollectingService#serve(ServiceRequestContext, HttpRequest)
```

Comment:
I'm not sure users will get anything useful from the Javadoc of these methods. How about referencing the builder methods?